## Expressions
values and operators - they all reduce to single value or answer. Order of operation rules apply

In [1]:
2+2

4

In [2]:
5-3

2

In [3]:
3*7

21

In [4]:
22/7

3.142857142857143

In [5]:
2+3*6

20

In [6]:
(2+3)*6

30

If type invalid expression, will get error message back.

## Data Types

Each value belongs only to 1 data type
Integers (no decimals)
Floating points (with decimals): 3.14 or 5.00
Strings: for text values. Always start and end with single quote. 'hello'.

Integers (no decimals)

Floating points (with decimals): 3.14 or 5.00

Strings: for text values. Always start and end with single quote. 'hello'. When using a plus operator with strings, it is concatenation of strings.

In [9]:
'Alice' + 'Bob'

'AliceBob'

#### String Replication

multiplication of strings - useful to make string repeat itself

In [10]:
'Alice' * 3

'AliceAliceAlice'

In [11]:
'Hello' + '!'*10

'Hello!!!!!!!!!!'

## Variables

Saving values to computer's memory. Stores a single value by assigning value to variable. Can rename or reassign values to the variable. This is called overwritting variable. The old one is forgotten.

In [12]:
spam = 42

In [13]:
spam

42

In [14]:
spam = 'Hello'

In [15]:
spam + ' World'

'Hello World'

## Statements

Don't evaluate to single value. Can have expression as part of it. If an instruction evaluates to single value --> it is expression. If not, it is a statement. Most generally just call them code or instruction anyways.

In [16]:
spam = 2+2

In [17]:
spam

4

In [21]:
spam = 10

In [22]:
spam

10

In [23]:
spam = spam + 1

In [24]:
spam

11